script/setup_heroku

```diff
@@ -1,45 +1,49 @@
-#!/bin/sh
-
-set -e
+#!/bin/bash
+
+set -ue
+
+function create_heroku_addon () {
+	if heroku addons:info "$1" &> /dev/null; then
+		echo "Already installed $1"
+	else
+		echo "Installing $1"
+		heroku addons:create "$@"
+	fi
+}
+
+if [ $# -eq 0 ]; then
+	echo >&2 "Please add your Heroku app name to the end of the command."
+	echo >&2 "Usage: setup_heroku your_app_name"
+	exit 1
+fi
 
-if [ $1 ]
-then
-  herokuApp="$1"
+export HEROKU_APP="$1"
 
-  echo "Getting ready to set environment variables for $herokuApp"
-  figaro heroku:set -e production --app $herokuApp
+echo "Getting ready to set environment variables for $HEROKU_APP"
+figaro heroku:set -e production
 
-  echo "Setting MAILER_URL"
-  heroku config:set MAILER_URL=$herokuApp.herokuapp.com --app $herokuApp
+echo "Setting MAILER_URL"
+heroku config:set MAILER_URL="$HEROKU_APP.herokuapp.com"
 
-  echo "Setting DEVISE_SECRET_KEY"
-  # generate a random string with 36 characters
-  token1=$(python -c 'import uuid; print (uuid.uuid4())')
-  heroku config:set DEVISE_SECRET_KEY=$token1 --app $herokuApp
+if [ -z "$(heroku config:get SECRET_TOKEN)" ]; then
+	echo "Setting SECRET_TOKEN"
+	# generate a random string with 36 characters
+	heroku config:set SECRET_TOKEN="$(ruby -rsecurerandom -e 'puts SecureRandom.uuid')"
```

Comment:
You're right, I did misread your comment, so thanks for straightening me out 🙈 👍  

I've pushed 2d7c28575ef81c67f3d6972cc57d1e8ed488c43f to run `bin/rake` instead of running `SecureRandom.uuid`.

I could also see a few other changes being made:

1. Call `figaro` with `bundle exec figaro` to ensure that the right version of `figaro` is being run
2. Call `bundle check` to check that `bundle` has been run
3. Check that `heroku` is installed with `type heroku`, then suggest running `brew install heroku` if `type brew` determines that Homebrew is available

Also, go ahead and let me know if you want any of these commits squashed.